pkg/controllers/syncer/testing/testing.go

```diff
@@ -34,6 +37,9 @@ type SyncTest struct {
 	ExpectedPhysicalState map[schema.GroupVersionKind][]runtime.Object
 	ExpectedVirtualState  map[schema.GroupVersionKind][]runtime.Object
 
+	ExpectNotFoundPhysical map[schema.GroupVersionKind][]types.NamespacedName
+	ExpectNotFoundVirtual  map[schema.GroupVersionKind][]types.NamespacedName
```

Comment:
Is this necessary? :thinking: 
I thought that if the fake client returns resources that are not in ExpectedPhysicalState( or ExpectedVirtualState) then the test will fail with a "... State mismatch ..." message. Some other test cases might already rely on this.

Reply:
It doesn't check unexpected objects. Just compares the state of expected ones. (code is identical for ExpectedVirtualState)

```
	// Compare states
	if s.ExpectedPhysicalState != nil {
		for gvk, objs := range s.ExpectedPhysicalState {
			err := compareObjs(t, s.Name+" physical state", ctx, pClient, gvk, scheme, objs, s.Compare)
			if err != nil {
				t.Fatalf("%s - Physical State mismatch: %v", s.Name, err)
			}
		}
	}
```

Reply:
Yes, you are right. But that would cover our use case if we declare that our expected state for some gvk is no resources.
The `compareObjs` that is called there^ first checks [if the object count matches](https://github.com/loft-sh/vcluster/blob/3a209556835064b76104223bfa0eee429c1f24ee/pkg/controllers/syncer/testing/testing.go#L113), and when it does it will look for every expected obj and [fail if it is not found](https://github.com/loft-sh/vcluster/blob/3a209556835064b76104223bfa0eee429c1f24ee/pkg/controllers/syncer/testing/testing.go#L173-L175).
So I think it should work as long as you include an empty map entry in the ExpectedVirtualState, e.g:
```
ExpectedVirtualState: map[schema.GroupVersionKind][]runtime.Object{
    storagev1.SchemeGroupVersion.WithKind(kind): {},
}
```
so this^ would verify that no resource of storagev1 `kind` exists in the virtual state.

Reply:
Changed to this :)